### PR TITLE
Add apt function to allow disabling home button

### PR DESCRIPTION
**Pull request by astronautlevel2**
Add a function to allow enabling/disabling whether the home button is usable.

This allows homebrew to block the home button, similar to the eShop when loading, MH4U when online, and almost certainly other games/applications which I can't think of off the top of my head.

Kind of a hack and we're almost certainly forgetting something, but it seems to work fine.

### CODE REVIEW

File: libctru/source/services/apt.c [hunk 1]
```diff
@@ -41,6 +41,7 @@ enum
 
 static u8 aptHomeButtonState;
 static u32 aptFlags = FLAG_ALLOWSLEEP;
+static bool home_allowed = false;
```
Comment (fincs) [inline libctru/source/services/apt.c]: Rename to aptHomeAllowed.

**Comment by fincs** [comment]
In addition, there should probably be an API to detect whether HOME is pressed and it is blocked. Official apps display a cute interdictory "forbidden HOME" logo in the bottom screen when this happens.

**Comment by astronautlevel2** [comment]
The function to check if HOME is blocked is already present - I can add the API to detect whether HOME is pressed tomorrow when I have a bit more time.

**Comment by fincs** [comment]
Any updates on this PR?

**Comment by fincs** [comment]
Pinging @astronautlevel2 again.

File: libctru/source/services/apt.c [hunk 2]
```diff
@@ -41,6 +41,7 @@ enum
 
 static u8 aptHomeButtonState;
 static u32 aptFlags = FLAG_ALLOWSLEEP;
+static bool aptHomeAllowed = false;
```
Comment (Oreo639) [inline libctru/source/services/apt.c]: I don't think it is a good idea to block going to the home menu by default.

Comment (Oreo639) [inline libctru/source/services/apt.c]: I don't think it is a good idea to block the home menu by default.

**Comment by fincs** [comment]
PR superseded by #430.

**Pull request closed without merging**